roles/ceph-rgw-loadbalancer/tasks/pre_requisite.yml

```diff
@@ -1,11 +1,58 @@
 ---
 - name: install haproxy and keepalived
   package:
-    name: ['haproxy', 'keepalived']
+    name: haproxy
     state: present
   register: result
   until: result is succeeded
 
+============ WIP
+
+- name: install keepalived
+  package:
+    name: keepalived
+    state: present
+  register: result
+  when: keepalived_repository == 'distro'
+  until: result is succeeded
+
+- name: configure keepalived custom repository
+  get_url:
+    url: "{{ keepalived_url }}"
+    dest: "/tmp/keepalived-{{ keepalived_version }}.tar.gz"
+    owner: root
+    group: root
+  when: keepalived_repository == 'version'
+
+- name: check if specified keepalived version already exists
+  shell: "ls /usr/sbin/ | grep keepalived-{{ keepalived_version }}"
+  register: check_keepalived_version
+
+- name: unarchive keepalived
+  unarchive:
+    src: "/tmp/keepalived-{{ keepalived_version }}.tar.gz"
+    dest /tmp/
+  when: check_keepalived_version.rc != 0
```

Comment:
I think this can all be completed with Ansible's [`unarchive`](https://docs.ansible.com/ansible/latest/modules/unarchive_module.html) module. Note that the examples include an https:// source.

Reply:
done